src/vs/workbench/browser/parts/editor/editorGroupsControl.ts

```diff
@@ -2027,10 +2037,34 @@ export class EditorGroupsControl extends Themable implements IEditorGroupsContro
 
 	private layoutEditor(position: Position): void {
 		const editorSize = this.silosSize[position];
-		if (editorSize && this.visibleEditors[position]) {
+		const editor = this.visibleEditors[position];
+		if (editorSize && editor) {
 			let editorWidth = this.layoutVertically ? editorSize : this.dimension.width;
 			let editorHeight = (this.layoutVertically ? this.dimension.height : this.silosSize[position]) - EditorGroupsControl.EDITOR_TITLE_HEIGHT;
 
+			let editorPosition = 0;
+
+			const autoActivate = this.configurationService.getValue<string>('centerMode.autoActivate');
+
+			let tryCentering =
+				autoActivate === 'fullscreen' && browser.isFullscreen() ||
+				autoActivate === 'zen' && this.partService.isInZenMode() ||
+				autoActivate === 'always';
+
+			if (tryCentering) {
+				const size = Math.max(100, this.configurationService.getValue<number>('centerMode.size'));
+				const adoptRight = this.configurationService.getValue<string>('centerMode.adoptRight');
+				const centerOnlyEditors = this.configurationService.getValue<string>('centerMode.onlyEditors');
+				const availableWidth = this.dimension.width;
+
+				const isTextEditor = editor instanceof TextResourceEditor;
```

Comment:
`instanceof` is not a good design... in fact it doesn't even work correct. Markdown editors and normal text editors seem to be of class type `TextFileEditor` (which I was not able to import because linter import rules).
Best option is to add another abstract method like `allowsCentering(): boolean` or something to `BaseEditor` and let the implementation decide whether it itself should be centered or not